imod/prepare/topsystem/allocation.py

```diff
@@ -134,6 +134,13 @@ def _is_layered(grid: GridDataArray):
     return "layer" in grid.sizes and grid.sizes["layer"] > 1
 
 
+def _enforce_layered_top(top: GridDataArray, bottom: GridDataArray):
+    if _is_layered(top):
+        return top
+    else:
+        return create_layered_top(bottom, top)
+
+
 @enforced_dim_order
 def _allocate_cells__stage_to_riv_bot(
     top: GridDataArray,
```

Comment:
for this function it would help to know which arrays can be time-dependent and which ones are constant.

Reply:
Extended the docstrings with this information